controllers/subscription/subscription_controller.go

```diff
@@ -0,0 +1,436 @@
+/*
+Copyright 2020 Red Hat.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	appsv1alpha1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	operatorConditions "github.com/operator-framework/api/pkg/operators/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/3scale/3scale-operator/controllers/subscription/csvlocator"
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/product"
+	"github.com/3scale/3scale-operator/pkg/helper"
+
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+	"github.com/3scale/3scale-operator/version"
+)
+
+// SubscriptionReconciler reconciles 3scale operator subscription object
+type SubscriptionReconciler struct {
+	*reconcilers.BaseReconciler
+	OperatorNamespace string
+}
+
+// blank assignment to verify that APIManagerReconciler implements reconcile.Reconciler
+var _ reconcile.Reconciler = &SubscriptionReconciler{}
+
+var (
+	systemredisRequirement  string
+	backendredisRequirement string
+	mysqlRequirement        string
+	postgresRequirement     string
+	rhtComponentVersion     string
+)
+
+const (
+	communityOperatorSubscription  = "3scale-community-operator"
+	threescaleOperatorSubscription = "3scale"
+	threescaleOperatorDeployment   = "threescale-operator-controller-manager-v2"
```

Comment:
```suggestion
	threescaleOperatorSubscription = "3scale"
```
Very minor but is there any reason to keep the unused constants?

Reply:
thanks!